src/hypergraph.jl

```diff
@@ -108,41 +134,100 @@ Returns hyperedges for a given vertex `v_id` in a hypergraph `h`.
 @inline gethyperedges(h::Hypergraph, v_id::Int) = h.v2he[v_id]
 
 """
-    add_vertex!(h::Hypergraph{T};hyperedges::Dict{Int,T} = Dict{Int,T}()) where T <: Real
+    add_vertex!(h::Hypergraph{T, V, E}; hyperedges::Dict{Int,T} = Dict{Int,T}(), vertex_val::Union{UndefInitializer, V} = undef) where {T <: Real, V, E}
 
 Adds a vertex to a given hypergraph `h`. Optionally, the vertex can be added
 to existing hyperedges. The `hyperedges` parameter presents a dictionary
-of hyperedge identifiers and values stored at the hyperedges
+of hyperedge identifiers and values stored at the hyperedges.
+Additionally, a value can be stored with the vertex using the `vertex_val` parameter.
 
 """
-function add_vertex!(h::Hypergraph{T};hyperedges::Dict{Int,T} = Dict{Int,T}()) where T <: Real
+function add_vertex!(h::Hypergraph{T, V, E}; hyperedges::Dict{Int,T} = Dict{Int,T}(), vertex_val::Union{UndefInitializer, V} = undef) where {T <: Real, V, E}
```

Comment:
1. make a line break
2. using `undef` here is a bad idea as its purpose in Julia is different. I would prefer `nothing`, and use the union with `Nothing` approach I have proposed above.

Reply:
Again - this is the behavior from `Array`s - to be discussed

Reply:
The behavior form `Array` has a different purpose. `undef` is introduced to avoid populating the array with initial values for performance reasons. It is used when you create thousands of arrays in hot code and populate their values based on some function, typically immediately after their creation. The reason is that e.g. `Vector{Float64, Nothing}` is less efficient than `Vector{Float64}` in linear algebra operations, so in such applications you might want to create an unitialized `Vector{Float64}`.

I would assume that our use case is different, as this metadata part is not a core of the functionality so it is better not to let `vs` etc. be filled with garbage. But I will not very strongly press here as long as everyone is clear what functionality we provide.